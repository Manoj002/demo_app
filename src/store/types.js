export const USER_REGISTERED = 'USER_REGISTERED';
export const SUBMIT_NAME = 'SUBMIT_NAME';
export const NAME_CHANGED = 'NAME_CHANGED';
