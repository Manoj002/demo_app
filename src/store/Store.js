import React, {createContext, useReducer} from 'react';
import {USER_REGISTERED, NAME_CHANGED, SUBMIT_NAME} from './types';

const initialState = {
  isUserRegistered: false,
  userName: undefined,
};

export const StoreContext = createContext(initialState);

export const reducer = (state, action) => {
  switch (action.type) {
    case USER_REGISTERED:
      return {...state, isUserRegistered: action.payload};

    case NAME_CHANGED:
      return {...state, userName: action.payload};

    case SUBMIT_NAME:
      return {...state, userName: action.payload};

    default:
      return state;
  }
};

export const StoreProvider = ({children}) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  return (
    <StoreContext.Provider value={[state, dispatch]}>
      {children}
    </StoreContext.Provider>
  );
};
