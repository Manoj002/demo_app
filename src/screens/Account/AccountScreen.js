import React, {useContext} from 'react';
import {Text} from '../../commonComponents';
import PropTypes from 'prop-types';
import {Container} from './AccountScreen.styles';
import {StoreContext} from '../../store/Store';
import Strings from '../../constants/Strings';

const AccountScreen = ({navigation}) => {
  const [state] = useContext(StoreContext);
  return (
    <Container>
      <Text
        align="right"
        fontSize={18}
        lineHeight={20}
        color="blue"
        marginRight={20}
        marginBottom={20}>
        {`${state.userName}'s ${Strings.ACCOUNT}`}
      </Text>
      <Text
        fontSize={18}
        lineHeight={20}
        color="white"
        align="center"
        marginTop={40}
        marginBottom={40}>
        {Strings.ACCOUNT_SCREEN}
      </Text>
      <Text
        marginBottom={28}
        color="blue"
        testID="HomeText"
        onPress={() => navigation.navigate('Home')}>
        {Strings.NAV_HOME}
      </Text>
      <Text
        testID="ProfileText"
        color="blue"
        onPress={() => navigation.navigate('Profile')}>
        {Strings.NAV_PROFILE}
      </Text>
    </Container>
  );
};

AccountScreen.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func,
  }).isRequired,
};

export default AccountScreen;
