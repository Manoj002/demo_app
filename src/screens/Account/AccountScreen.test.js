import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import AccountScreen from './AccountScreen';
import {StoreProvider, StoreContext} from '../../store/Store';

const navigation = {
  navigate: jest.fn(),
};

const mockValues = {
  isUserRegistered: true,
  userName: 'Manoj',
};

const mockDispatch = jest.fn();

describe('Account Screen', () => {
  beforeEach(() => jest.clearAllMocks());

  it('renders correctly', () => {
    const Container = render(
      <StoreProvider>
        <StoreContext.Provider value={[mockValues, mockDispatch]}>
          <AccountScreen navigation={navigation} />
        </StoreContext.Provider>
      </StoreProvider>,
    );
    expect(Container).toMatchSnapshot();
  });

  it('should navigate to ProfileScreen on Text Press', () => {
    const Container = render(
      <StoreProvider>
        <StoreContext.Provider value={[mockValues, mockDispatch]}>
          <AccountScreen navigation={navigation} />
        </StoreContext.Provider>
      </StoreProvider>,
    );
    const {getByTestId} = Container;
    const ProfileText = getByTestId('ProfileText');
    fireEvent(ProfileText, 'onPress');
    expect(navigation.navigate).toHaveBeenCalledTimes(1);
    expect(navigation.navigate).toHaveBeenCalledWith('Profile');
  });

  it('should navigate to HomeScreen on Text Press', () => {
    const Container = render(
      <StoreProvider>
        <StoreContext.Provider value={[mockValues, mockDispatch]}>
          <AccountScreen navigation={navigation} />
        </StoreContext.Provider>
      </StoreProvider>,
    );
    const {getByTestId} = Container;
    const HomeText = getByTestId('HomeText');
    fireEvent(HomeText, 'onPress');
    expect(navigation.navigate).toHaveBeenCalledTimes(1);
    expect(navigation.navigate).toHaveBeenCalledWith('Home');
  });
});
