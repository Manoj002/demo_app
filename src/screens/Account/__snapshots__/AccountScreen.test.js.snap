// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Screen renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#212124",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
      },
    ]
  }
>
  <Text
    align="right"
    bold={false}
    color="blue"
    fontSize={18}
    lineHeight={20}
    marginBottom={20}
    marginLeft={0}
    marginRight={20}
    marginTop={0}
    style={
      Array [
        Object {
          "color": "#007AFF",
          "fontFamily": "Roboto",
          "fontSize": 18,
          "lineHeight": 20,
          "marginBottom": 20,
          "textAlign": "right",
        },
      ]
    }
  >
    Manoj's Account
  </Text>
  <Text
    align="center"
    bold={false}
    color="white"
    fontSize={18}
    lineHeight={20}
    marginBottom={40}
    marginLeft={0}
    marginTop={40}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Roboto",
          "fontSize": 18,
          "lineHeight": 20,
          "marginBottom": 40,
          "marginTop": 40,
          "textAlign": "center",
        },
      ]
    }
  >
    This is Account Screen
  </Text>
  <Text
    align="left"
    bold={false}
    color="blue"
    fontSize={16}
    lineHeight={18}
    marginBottom={28}
    marginLeft={0}
    marginTop={0}
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": "#007AFF",
          "fontFamily": "Roboto",
          "fontSize": 16,
          "lineHeight": 18,
          "marginBottom": 28,
          "textAlign": "left",
        },
      ]
    }
    testID="HomeText"
  >
    Go to Home
  </Text>
  <Text
    align="left"
    bold={false}
    color="blue"
    fontSize={16}
    lineHeight={18}
    marginBottom={0}
    marginLeft={0}
    marginTop={0}
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": "#007AFF",
          "fontFamily": "Roboto",
          "fontSize": 16,
          "lineHeight": 18,
          "textAlign": "left",
        },
      ]
    }
    testID="ProfileText"
  >
    Go to Profile
  </Text>
</View>
`;
